File: doc-examples/example-jackson-cbor-java/src/test/java/example/BookHttpTest.java
package example;

import io.micronaut.test.extensions.junit5.annotation.MicronautTest;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertNotNull;

@MicronautTest
class BookHttpTest {

    @Test
    void cborHttpRoundTrip(BookClient client) {
        Book saved = client.save(new Book("The Stand", 50));
        assertNotNull(saved);
        assertEquals("The Stand", saved.getTitle());
        assertEquals(50, saved.getQuantity());
    }
}

File: doc-examples/example-jackson-cbor-kotlin/src/test/kotlin/example/BookHttpTest.kt
package example

import io.micronaut.test.extensions.junit5.annotation.MicronautTest
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Assertions.assertNotNull
import org.junit.jupiter.api.Test

@MicronautTest
class BookHttpTest {

    @Test
    fun cborHttpRoundTrip(client: BookClient) {
        val saved = client.save(Book("The Stand", 50))
        assertNotNull(saved)
        assertEquals("The Stand", saved.title)
        assertEquals(50, saved.quantity)
    }
}
